ext/hal/cmsis/Kconfig

```diff
@@ -15,3 +15,14 @@ config HAS_CMSIS_CORE_R
 	bool
 
 endif
+
+menuconfig HAS_CMSIS_DSP
```

Comment:
Wonder if this could be `select`ed from some SOC symbol. Does it make sense to turn it off in menuconfig?

Reply:
Unlike the CMSIS-Core component, which is an essential processor interface driver that is selected by the arch symbol, the CMSIS-DSP component is an optional library and therefore should be selected by the component/library/application that uses it.

Also it is worth noting that the CMSIS-DSP library can be used on the processor cores with and without the DSP extension.

Reply:
Ah... might've been the naming that confused me. `HAS_...` to me sounds like it's describing something that's either available or not, and not an optional configurable feature.